### PR TITLE
RHMAP-12406 - FHC tool - Check if the tag is valid or not before doin…

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/RHMAP-12406
Change: Make a validation for the tag value as it is made into the studio. ( don't allow builds with used tag connections ) 

Following the local tests to check.

**Error of this validation(changes)**

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js build project=qei6x6jgfwgrrzpi4v6ifuej app=qei6x6nrkiamo6frercbc4ps cloud_app=qei6x6pzt7tne5f2gx2wootn environment=dev destination=android tag=4.0.3
(node) sys is deprecated. Use util instead.
fhc ERR! This tag is currently in use, please choose another.
fhc Command executed with error
```
**Success with tag and changes**

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js build project=qei6x6jgfwgrrzpi4v6ifuej app=qei6x6nrkiamo6frercbc4ps cloud_app=qei6x6pzt7tne5f2gx2wootn environment=dev destination=android tag=5.0.3
(node) sys is deprecated. Use util instead.
.......................
Download URL: https://support.us.feedhenry.com/digman/android-v3/dist/5a8a6d97-dd5b-43ce-a07a-4b119bb63b1d/android~4.0~61~CordovaApp.apk?digger=diggers.sam1-farm2-linux1
```
**Sucess without tag**

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js build project=qei6x6jgfwgrrzpi4v6ifuej app=qei6x6nrkiamo6frercbc4ps cloud_app=qei6x6pzt7tne5f2gx2wootn environment=dev destination=android 
(node) sys is deprecated. Use util instead.
...........
Download URL: https://support.us.feedhenry.com/digman/android-v3/dist/fbb39f88-7e7f-4461-b384-46ead9107136/android~4.0~62~CordovaApp.apk?digger=diggers.sam1-farm2-linux1
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

### CODE REVIEW

File: lib/cmd/common/build.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ function build(argv, cb) {
     validateArgs(argObj);
     if (isNGUI()) {
       validateArgsNgui(argObj);
+      checkTag(argObj,cb);
```
Comment (wei-lee) [inline lib/cmd/common/build.js]: the `checkTag` is an async function call. If it's invoked like this, the functions from L53-54 will also be invoked as well, even if `checkTag` is failed. 

The right way to do it is to call the next 2 functions only if `checkTag` is passed,  something like this:

```javascript
checkTag(argObj, function(err){
  if (err) {
    return cb(err);
  } else {
     mutateGitRefArgs(argObj);
     doBuild(argObj, cb);
  }
});
```

Comment (camilamacedo86) [inline lib/cmd/common/build.js]: I got the idea @wei-lee . Tks for the review .. I will change it.

**Comment by camilamacedo86** [comment]
@wei-lee . I changed the PR for the error be exactly equals all other errors that have in the build.js 
We need one-day refectory this command for V3 format and this outputs.

**Comment by camilamacedo86** [comment]
@wei-lee changes requested made. Please, could you check it now?

**Comment by wei-lee** [comment]
@camilamacedo86 Your change is ok now, but I just think about the fix again and I am not sure list all the tags for every build request is a good idea. I know some projects in production could have many connection tags and request listing all of them for each build seems like a overkill. I think the best option is here is actually to do it on the server as part of the build request validation - it can easily check if the tag is an existing one and return error if it is.

Sorry I should have thought of it earlier, but I hope it makes sense.

**Comment by camilamacedo86** [comment]
@wei-lee I agree with you. I will change the JIRA for this fix be in the back/Rest Service and I am closing this PR.

**Pull request closed without merging**